src/d3d9/d3d9_common_buffer.cpp

```diff
@@ -111,12 +111,20 @@ namespace dxvk {
       info.stages |= VK_PIPELINE_STAGE_HOST_BIT;
       info.access |= VK_ACCESS_HOST_WRITE_BIT;
 
-      if (!(m_desc.Usage & D3DUSAGE_WRITEONLY))
-        info.access |= VK_ACCESS_HOST_READ_BIT;
-
       memoryFlags |= VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT
-                  |  VK_MEMORY_PROPERTY_HOST_COHERENT_BIT
-                  |  VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT;
+                  |  VK_MEMORY_PROPERTY_HOST_COHERENT_BIT;
+
+      if ((m_desc.Usage & (D3DUSAGE_WRITEONLY | D3DUSAGE_DYNAMIC)) == 0
+        || DoPerDrawUpload()
+        || m_parent->CanSWVP()
```

Comment:
This may introduce a performance regression for MIXED mode HWVP games (which aren't that uncommon). I'd recommend bringing back CanOnlySWVP() to check for D3DCREATE_SOFTWARE_VERTEXPROCESSING exclusively and using it here. 

We shouldn't hit MIXED mode SWVP paths with dxvk anyway, because if a game sees it can HWVP in MIXED mode, it most likely will.